lib/src/badge/Badge.tsx

```diff
@@ -1,47 +1,156 @@
 import React from "react";
-import styled, { ThemeProvider } from "styled-components";
-import useTheme from "../useTheme";
+import styled from "styled-components";
 import BadgePropsType from "./types";
+import DxcFlex from "../flex/Flex";
+import CoreTokens from "../common/coreTokens";
 
-const DxcBadge = ({ notificationText, disabled }: BadgePropsType): JSX.Element => {
-  const colorsTheme = useTheme();
+const contextualColorMap = {
+  grey: {
+    background: CoreTokens.color_grey_200,
+    text: CoreTokens.color_grey_900,
+  },
+  blue: {
+    background: CoreTokens.color_blue_200,
+    text: CoreTokens.color_blue_900,
+  },
+  green: {
+    background: CoreTokens.color_green_200,
+    text: CoreTokens.color_green_900,
+  },
+  orange: {
+    background: CoreTokens.color_orange_200,
+    text: CoreTokens.color_orange_900,
+  },
+  red: {
+    background: CoreTokens.color_red_200,
+    text: CoreTokens.color_red_900,
+  },
+  yellow: {
+    background: CoreTokens.color_yellow_200,
+    text: CoreTokens.color_yellow_900,
+  },
+  purple: {
+    background: CoreTokens.color_purple_200,
+    text: CoreTokens.color_purple_900,
+  },
+};
+
+const sizeMap = {
+  small: {
+    height: "20px",
+    width: "20px",
+    minWidth: "20px",
+    fontSize: CoreTokens.type_scale_01,
+    borderRadius: CoreTokens.spacing_12,
+    iconSize: "14px",
+    padding: {
+      contextual: `${CoreTokens.spacing_4}`,
+      notification: `${CoreTokens.spacing_0} ${CoreTokens.spacing_4}`,
+    },
+  },
+  medium: {
+    height: "24px",
+    width: "24px",
+    minWidth: "24px",
+    fontSize: CoreTokens.type_scale_02,
+    borderRadius: CoreTokens.spacing_12,
+    iconSize: "16px",
+    padding: {
+      contextual: `${CoreTokens.spacing_4} ${CoreTokens.spacing_8}`,
+      notification: `${CoreTokens.spacing_0} ${CoreTokens.spacing_4}`,
+    },
+  },
+  large: {
+    height: "32px",
+    width: "32px",
+    minWidth: "32px",
+    fontSize: CoreTokens.type_scale_04,
+    borderRadius: CoreTokens.spacing_24,
+    iconSize: "24px",
+    padding: {
+      contextual: `${CoreTokens.spacing_4} ${CoreTokens.spacing_8}`,
+      notification: `${CoreTokens.spacing_0} ${CoreTokens.spacing_8}`,
+    },
+  },
+};
+
+const Label = ({ label, notificationLimit, size }) => {
+  return (
+    <LabelContainer size={size}>
+      {typeof label === "number" ? (label > notificationLimit ? "+" + notificationLimit : label) : label}
+    </LabelContainer>
+  );
+};
+
+const DxcBadge = ({
+  label,
+  title,
+  mode = "contextual",
+  color = "grey",
```

Comment:
The default value of the prop color should be calculated according to the mode (`color = mode === "contextual" ? "grey" : "red"`), if not, when mode ="notification" color would be 'grey' which is strange since they are red.

![image](https://github.com/dxc-technology/halstack-react/assets/67998605/84d18c51-887c-4ebd-beea-be77cb45463e)

Reply:
I limited the color prop to only contextual mode. Notification mode will not have color prop.